### PR TITLE
Give `reim` some names

### PR DESCRIPTION
**Pull request by fjebaker**
Details in the commit messages.

### CODE REVIEW

File: subroutines/genreltrans.f90 [hunk 1]
```diff
@@ -135,9 +135,9 @@ subroutine config_frequency(config, model_args)
 
         ! Convert frequency bounds from Hz to c/Rg (now being more accurate with
         ! constants)
-        config%fhi = dble(model_args%fhiHz) * 4.92695275718945d-06 *           &
+        config%fhi = dble(model_args%fhiHz) * 4.92695275718945d *              &
```
Comment (fjebaker) [inline subroutines/genreltrans.f90]: Whoops that's very alarming behaviour from the formatter let me fix that.

**Review by mgullik — COMMENTED** [review]
Hey @fjebaker, this is great! 
A couple of things: 
- I merged #139, so we need to add the case of ReIm = 8 
-#139 restyled the tree decision on the output, you could have seen from the conflict but I wanted to post it out since it's one of the biggest change that concerns this PR 
- ReIm can be also negative, we designed these cases to have the possibilities of looking at the lag predicted by the model without folding into a response matrix. In genreltrans there is a IF statement that checks if ReIm .gt. 0, if not propercross_NOmatrix is called instead of propercross. This is useful only for ReIm=-3 and ReIm=-4, because using ReIm=-5 and -6 prompts to the response matrix anyway.

**Comment by fjebaker** [comment]
> ReIm can be also negative

Aaah, I was wondering why there were `abs` scattered about the place (Edit: on looking at the code again, it's actually abvious what those `.gt. 0 ` checks were doing. I couldn't see the forest for the trees!). I'd based my changes on what I'd read here:

https://reltransdocs.readthedocs.io/en/latest/model_flavours.html

It seems a little confusing to have a seperate reim for folded lag energy but then have others just be their negatives... we should add some tests for negative `reim` then, and document that properly somewhere. For this PR, I'm tempted to remap internally `5` to `-3` and so on, so that the enumeration is

```
! The time-averaged spectrum.
RE_IM_SPECTRUM                    =  0,
RE_IM_CROSS_SPEC_REAL             =  1,
RE_IM_CROSS_SPEC_REAL_FOLDED      = -1,
RE_IM_CROSS_SPEC_IMAG             =  2,
RE_IM_CROSS_SPEC_IMAG_FOLDED      = -2,
RE_IM_CROSS_SPEC_MODULUS          =  3,
RE_IM_CROSS_SPEC_MODULUS_FOLDED   = -3,
RE_IM_LAG_ENERGY                  =  4,
RE_IM_LAG_ENERGY_FOLDED           = -4,
RE_IM_LAG_FREQ                    =  7
```

and we then have a function that parses the old `reim` to the new value system, so that calling `reim=5` and `reim=-3` is the same.

Would that be alright?

**Comment by mgullik** [comment]
Hi @fjebaker, sorry my explanation was really bad and I also wrote something incorrect. 
I was wrong saying that ReIm = -3 and ReIm = -4 are the only useful ones, also ReIm = -1 and ReIm = -2 are useful since if we use ReIm = [1,2,3,4] and fmin != 0 and fmax != 0 the code wants the telescope response. So ReIm = [-1, -2, -3, -4] makes sense. 
What it doesn't make sense is ReIm = [-5, -6] because even though we use propercross_NOmatrix, before the output we call cfoldandbin on the cross spectrum which folds into the telescope response anyway. 

So, let me try to be more clear:  
- making reim=5 and reim=-3 equal is not correct. 
- the correct possibilities are ReIm = [-1, -2, -3, -4, 1, 2, 3 , 4, 5, 6, 7, 8], so the boundaries of the ReIm parameter should be (-4 to 8) 
- the positive options are when we fold, so your table should look like this 
RE_IM_SPECTRUM                                              =  0,
RE_IM_CROSS_SPEC_REAL_FOLDED              =  1,
RE_IM_CROSS_SPEC_REAL                               = -1,
RE_IM_CROSS_SPEC_IMAG_FOLDED             =  2,
RE_IM_CROSS_SPEC_IMAG                              = -2,
RE_IM_CROSS_SPEC_MODULUS_FOLDED    =  3,
RE_IM_CROSS_SPEC_MODULUS                     = -3,
RE_IM_LAG_ENERGY_FOLDED                         =  4,
RE_IM_LAG_ENERGY                                          = -4,
RE_IM_CROSS_SPEC_MODULUS                     =  5,
RE_IM_LAG_ENERGY_FOLDED                         =  6,
RE_IM_LAG_FREQ                                               =  7, 
RE_IM_LAG_TWO_RESPONSES                        = 8

- you are totally right that it is not very intuitive to have negative options as not folded versions of the positive options, but negative options is just for us   

p.s. I merged the #139 so you should be to implement ReIm=8

**Comment by fjebaker** [comment]
End of day on Friday, I misread what you'd written. So negative is **not folded**, whereas positive is **folded**. But in that case `5` maps to `3` should be correct?

I am also slightly confused in that I've done the exact opposite but the tests still pass... Perhaps we look at this on the first day of the hackathon and clear up the difference and add `reim==8` together then.

**Comment by fjebaker** [comment]
Following on from the developer call on 2026-06-15: use terminology like "reference band folded" or "subject band folded".

File: subroutines/constants.f90 [hunk 2]
```diff
@@ -0,0 +1,103 @@
+#ifndef _RELTRANS_IS_DEBUG
+#define _RELTRANS_IS_DEBUG 0
+#endif
+
+module rtconstants
+! This module defines constants and enumerations used throughout the reltrans
+! codebase.
+! Physical constants should not be defined here, as a constants module already
+! exists in `amodules.f90`.
+! TODO: move `constants` in `amodules.f90` here.
```
Comment (mgullik) [inline subroutines/constants.f90]: We haven't set up any code convention for docstring in the modules. 
Should we make this description as in docstrings !>

File: subroutines/genreltrans.f90 [hunk 3]
```diff
@@ -608,59 +609,52 @@ subroutine genreltrans(Cp, dset, nlp, ear, ne, param, ifl, photar)
...
-
-    case(2)          !Imaginary part
+    else if (is_mode(model_args%reim, MODE_CROSS_SPEC_IMAG)) then
        photar = ImS
-
-    case(3,5)        !Modulus
+    else if (is_mode(model_args%reim, MODE_CROSS_SPEC_MODULUS)) then
        photar = sqrt(ReS**2 + ImS**2)
-       if (model_args%ReIm==3) then 
+       if (model_args%ReIm==3) then
```
Comment (mgullik) [inline subroutines/genreltrans.f90]:
```suggestion
       if (model_args%reim == MODE_CROSS_SPEC_MODULUS) then
```

File: subroutines/genreltrans.f90 [hunk 4]
```diff
@@ -608,59 +609,52 @@ subroutine genreltrans(Cp, dset, nlp, ear, ne, param, ifl, photar)
...
+    !Generate the correct output
+    if (is_mode(model_args%reim, MODE_CROSS_SPEC_REAL)) then
        photar = ReS
-
-    case(2)          !Imaginary part
+    else if (is_mode(model_args%reim, MODE_CROSS_SPEC_IMAG)) then
        photar = ImS
-
-    case(3,5)        !Modulus
+    else if (is_mode(model_args%reim, MODE_CROSS_SPEC_MODULUS)) then
```
Comment (mgullik) [inline subroutines/genreltrans.f90:637]: This IF statement should be true even when reim is 5, thus when reim is MODE_CROSS_SPEC_MODULUS_BOTH_FOLDED.
This works because we do not have test for reim == MODE_CROSS_SPEC_MODULUS_BOTH_FOLDED

File: subroutines/constants.f90 [hunk 5]
```diff
@@ -0,0 +1,103 @@
...
+        MODE_CROSS_SPEC_MODULUS_BOTH_FOLDED =  5,                              &
+        MODE_CROSS_SPEC_LAG_BOTH_FOLDED     =  6,                              &
+        MODE_LAG_FREQ                       =  7,                              &
+        MODE_CROSS_SPEC_LAG_TWO_RESPONSES   =  8
+
+contains
+
+    logical function is_mode(reim, mode) result(ret)
+        !> Checks whether `reim` is of a particular mode regardless of whether
+        !it is > folded or not.
```
Comment (mgullik) [inline subroutines/constants.f90]: Just a typo
```suggestion
        !> it is folded or not.
```

File: subroutines/constants.f90 [hunk 6]
```diff
@@ -0,0 +1,103 @@
...
+        select case(reim)
+        case (-4:8)
+            ret = reim
+        case default
+            print *, "FATAL: Invalid reim parameter: ", reim
+            stop 1
+        end select
+    end function parse_reim
+
+    logical function is_folded(reim) result(bool)
```
Comment (mgullik) [inline subroutines/constants.f90]: I don't think we even fold the subject without folding the reference band, thus this function might be called is_refband_folded or is_ref_folded

File: subroutines/genreltrans.f90 [hunk 7]
```diff
@@ -608,59 +609,52 @@ subroutine genreltrans(Cp, dset, nlp, ear, ne, param, ifl, photar)
...
        end if
-
-    case(4,6)        !Time lag
+    else if (is_mode(model_args%reim, MODE_CROSS_SPEC_LAG)) then
        do i = 1, ne
           dE = ear(i) - ear(i-1)
           photar(i) = atan2(ImS(i), ReS(i)) / (2.0*pi*config%fc) * dE
        end do
-       if (model_args%ReIm==4) then 
+       if (model_args%ReIm==4) then
```
Comment (mgullik) [inline subroutines/genreltrans.f90]:
```suggestion
       if (model_args%ReIm==MODE_CROSS_SPEC_LAG_REF_FOLDED) then
```

File: subroutines/genreltrans.f90 [hunk 8]
```diff
@@ -608,59 +609,52 @@ subroutine genreltrans(Cp, dset, nlp, ear, ne, param, ifl, photar)
...
-    case(3,5)        !Modulus
+    else if (is_mode(model_args%reim, MODE_CROSS_SPEC_MODULUS)) then
        photar = sqrt(ReS**2 + ImS**2)
-       if (model_args%ReIm==3) then 
+       if (model_args%ReIm==3) then
           write(*, *) "Warning ReIm = 3 should not be used for fitting!"
        end if
-
-    case(4,6)        !Time lag
+    else if (is_mode(model_args%reim, MODE_CROSS_SPEC_LAG)) then
```
Comment (mgullik) [inline subroutines/genreltrans.f90:642]: This has the same problem of the MODULUS above. It should return true even when reim == MODE_CROSS_SPEC_LAG_BOTH_FOLDED

**Review by mgullik — COMMENTED** [review]
Hi @fjebaker, thanks for the PR, I left a bunch of comments. 
One thing we should check is where in the code reim == 0 happens, and also think if we really need reim == 0. The time averaged call should be just with fmin == fmax == 0

**Review by mgullik — APPROVED** [review]
🦑

**Review by mgullik — APPROVED** [review]
🦑
